htdocs/core/customreports.php

```diff
@@ -56,14 +61,10 @@
 	//$search_xaxis = GETPOST('search_xaxis', 'array');
 	if (GETPOST('search_xaxis', 'alpha') && GETPOST('search_xaxis', 'alpha') != '-1') {
 		$search_xaxis = array(GETPOST('search_xaxis', 'alpha'));
-	} else {
```

Comment:
Why removing this else ?

Reply:
When the customreport.php page is called as a standalone page, i do not see at wich line the $search_xaxis is already initialized. 
Can you point where the init is done ?

Reply:
In the "else" branch where this value is not set, we have explicitly this comment:
```
    // $search_measures, $search_xaxis or $search_yaxis may have been defined by the parent.
```

It says 'may be', so the code the follows needs to be protected from unset variables.

We can observe this protection:
https://github.com/Dolibarr/dolibarr/blob/28a955204f963054a93e39f32f8e3329efd72d30/htdocs/core/customreports.php#L410-L413

https://github.com/Dolibarr/dolibarr/blob/28a955204f963054a93e39f32f8e3329efd72d30/htdocs/core/customreports.php#L410-L413

Therefore it is useless to initialize variables to "empty values" that are protected with isset anyway.

In case the protection is unsufficient, the static analysis should kick in and report it.

Reply:
I think you are right.
It's true. value can be set by parent when the page is called as an include. So should not be erased in this case.
When called as a standalone page, it may be initialized but if there is a protection later, we can avoid it to simplify code.
Will wait CTI green to merge with your removal.